sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/fileSourceInterfaces.scala

```diff
@@ -291,8 +291,12 @@ class HDFSFileCatalog(
   refresh()
 
   override def listFiles(filters: Seq[Expression]): Seq[Partition] = {
+
     if (partitionSpec().partitionColumns.isEmpty) {
-      Partition(InternalRow.empty, allFiles().filterNot(_.getPath.getName startsWith "_")) :: Nil
+      Partition(
+        InternalRow.empty,
+        unpartitionedDataFiles().filterNot(_.getPath.getName startsWith "_")
```

Comment:
I dont know for sure. if there is a partitioning scheme, there is an additional filter on files that start with "_" in `listFiles`, that does not seem to be present in `allFiles`. So I am not sure where its best to merge.

Also, I think this way is slightly cleaner than listFiles conditionally depending on allFiles.